tests/Unit/Core/Compiler/NodeConverterTest.php

```diff
@@ -100,37 +100,52 @@ public function getConvertTestValues()
         $multiRoot->addChildNode(new TextNode('bar'));
         $multiRoot->addChildNode(new TextNode('baz'));
         return [
-            [
+            'Reference to all variable converts to getAll() call' => [
                 new ObjectAccessorNode('_all'),
                 '$renderingContext->getVariableProvider()->getAll()'
             ],
-            [
+            'Reference to foo.bar converts to getByPath() call' => [
                 new ObjectAccessorNode('foo.bar'),
                 '$renderingContext->getVariableProvider()->getByPath(\'foo.bar\', $array0)'
             ],
-            [
+            'Reference to foo.bar with array accessors converts to array-access inline statement' => [
                 new ObjectAccessorNode('foo.bar', ['array', 'array']),
                 'isset($renderingContext->getVariableProvider()[\'foo\'][\'bar\']) ? $renderingContext->getVariableProvider()[\'foo\'][\'bar\'] : NULL'
             ],
-            [
+            'Boolean node with text TRUE converts to boolean true' => [
                 new BooleanNode(new TextNode('TRUE')),
-                'TYPO3Fluid\Fluid\Core\Parser\SyntaxTree\BooleanNode::convertToBoolean(
-					$expression1(
-						TYPO3Fluid\Fluid\Core\Parser\SyntaxTree\BooleanNode::gatherContext($renderingContext, $array0)
-					),
-					$renderingContext
-				)'
+                'true'
+            ],
+            'Boolean node with text FALSE converts to boolean false' => [
+                new BooleanNode(new TextNode('FALSE')),
+                'false'
+            ],
+            'Boolean node with text 1 converts to boolean true' => [
+                new BooleanNode(new TextNode('1')),
+                'true'
+            ],
+            'Boolean node with text 0 converts to boolean false' => [
+                new BooleanNode(new TextNode('0')),
+                'false'
+            ],
+            'Boolean node with number 1 converts to boolean true' => [
+                new BooleanNode(new NumericNode(1)),
+                'true'
+            ],
+            'Boolean node with number 0 converts to boolean false' => [
+                new BooleanNode(new NumericNode(0)),
+                'false'
```

Comment:
Can we add one more case for the empty string to make sure this implicit behavior cannot be broken?

Reply:
Sure!